frontend/src/metabase/common/components/EntityPicker/components/EntityPickerModal/EntityPickerModal.tsx

```diff
@@ -0,0 +1,125 @@
+import { useState, useMemo } from "react";
+import { t } from "ttag";
+
+import ErrorBoundary from "metabase/ErrorBoundary";
+import { useModalOpen } from "metabase/hooks/use-modal-open";
+import { Modal } from "metabase/ui";
+import type { SearchResult } from "metabase-types/api";
+
+import type {
+  EntityPickerOptions,
+  EntityTab,
+  CollectionPickerItem,
+} from "../../types";
+import { EntityPickerSearchInput } from "../EntityPickerSearch/EntityPickerSearch";
+
+import { ButtonBar } from "./ButtonBar";
+import { GrowFlex, ModalContent, ModalBody } from "./EntityPickerModal.styled";
+import { SinglePickerView } from "./SinglePickerView";
+import { TabsView } from "./TabsView";
+
+export type EntityPickerModalOptions = {
+  showPersonalCollection?: boolean;
+  showRootCollection?: boolean;
+  showSearch?: boolean;
+  hasConfirmButtons?: boolean;
+  allowCreateNew?: boolean;
+};
+
+export const defaultOptions: EntityPickerModalOptions = {
+  showPersonalCollection: true,
+  showRootCollection: true,
+  showSearch: true,
+  hasConfirmButtons: true,
+  allowCreateNew: true,
+};
+
+export interface EntityPickerModalProps {
+  title: string;
+  selectedItem: CollectionPickerItem | null;
+  onConfirm: () => void;
+  onItemSelect: (item: CollectionPickerItem) => void;
+  onClose: () => void;
+  tabs: EntityTab[];
+  options?: EntityPickerOptions;
+  searchResultFilter?: (results: SearchResult[]) => SearchResult[];
+  actions?: JSX.Element[];
+  trapFocus?: boolean;
+}
+
+export function EntityPickerModal({
+  title = t`Choose an item`,
+  onItemSelect,
+  onConfirm,
+  selectedItem,
+  onClose,
+  tabs,
+  options = defaultOptions,
```

Comment:
I wonder if we should spread in default options to provided options. if we get a partial options object things might break.

Reply:
Yeah, I like that idea